src/cls/rbd/cls_rbd_client.cc

```diff
@@ -537,6 +537,8 @@ namespace librbd {
         op->exec("rbd", "get_parent", bl3);
         ::encode(snap_id, bl4);
         op->exec("rbd", "get_protection_status", bl4);
+        ::encode(snap_id, bl5);
```

Comment:
Since this is a new function -- you need to query it as a separate step. Otherwise, newer librbds talking to older OSDs would fail the entire transaction.